crates/aptos-transactions-filter/src/transaction_matcher.rs

```diff
@@ -33,24 +33,32 @@ pub enum Matcher {
 impl Matcher {
     fn matches(
         &self,
-        block_id: HashValue,
-        block_epoch: u64,
-        block_timestamp: u64,
+        block_id: Option<HashValue>,
```

Comment:
it feels we should split the block related filter and txn based ones and make them composable

Reply:
Yeah, I did wonder about this. My only concern is breaking backward compatibility (e.g., the API has already made this feature public for a while, and it's kinda hard to tell how it's already being used). If we're okay breaking compatibility, I can split the filter up (e.g, txn and block rules)? Let me know 😄

Reply:
Gotcha. In that case, I'll build another PR on top of the entire stack, as the refactors are quite large.